Add missing modules to feature profiles

Johennes opened the pull request:
I also sorted the table by section occurrence and copied the section title in cases where it differed from the title in the table.

Fixes: #1437 

### Pull Request Checklist

<!-- Please read CONTRIBUTING.rst before submitting your pull request -->

* [ ] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

turt2live reviewed (changes requested):
Thanks! I'd suggest we split out the adding and resorting to their own PRs. It's not immediately clear to me that this is the ordering we'd want (previously it was loosely ordered on "required-ness"). Adding the missing stuff is certainly important, though.

Johennes commented:
Ok, I can do that. Where would you suggest I insert the missing modules in the existing order?

turt2live commented:
We typically just append them to the end, for better or worse. If they're "required" (they shouldn't be), put them under the existing required ones.

Johennes commented:
I pushed #1860 which merely adds the three missing modules without touching the table ordering.

Johennes commented:
The brief discussion in the matrix-spec room was, sadly, inconclusive. The suggested options so far are:

1. Sort alphabetically
2. Sort by section occurrence
3. Sort by requiredness (this is more or less how the table is sorted currently)
4. Sort by requiredness first and alphabetically second

As a middleground, I updated the PR to match option 4 but also renamed those rows where the module name didn't match the section title. This should hopefully still make it easy enough to cross-reference the table with the sections.

For convenience, the PR can be reviewed by commit:

* https://github.com/matrix-org/matrix-spec/pull/1855/commits/aa93f8b7126cb44ea8fec9a44543c5a6440155ed does the renaming
* https://github.com/matrix-org/matrix-spec/pull/1855/commits/5afe6ed19f2559d1567f02c4be2d3eb38a852eab does the sorting

Johennes commented:
CC @uhoreg / @anoadragon453 since you participated in the chat in matrix-spec.

turt2live reviewed (commented):
I'm going to abstain from reviewing this one - the order doesn't bother me.
